### ISSUE TITLE
Ogopogo error spam

### ISSUE THREAD

**Issue by shavedmonkey69** [BiodiversityLC/BiodiversityPublic#2]
I was driving the company cruiser near the water in Adamance. The Ogopogo then tried to eat me but couldn't because I was in the cruiser. After this the console was being spammed by this error message (at least 5-7minutes) before I quit the game.

`[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
Biodiversity.Creatures.Ogopogo.VerminAI.WaterTop (UnityEngine.BoxCollider coll) (at ./Creatures/Ogopogo/VerminAI.cs:246)
Biodiversity.Creatures.Ogopogo.VerminAI.DoAIInterval () (at ./Creatures/Ogopogo/VerminAI.cs:268)
EnemyAI.Update () (at <d0d093e9162748b6a00ca8938a14b439>:IL_048B)
Biodiversity.Creatures.Ogopogo.VerminAI.Update () (at ./Creatures/Ogopogo/VerminAI.cs:202)`

**Comment by NotAVerb** [comment]
I also have encountered Ogopogo logspam. Though this may have a different cause. I saw the Ogopogo eat a LethalInternship intern off in the distance on Nikki's Gorgonzola, so it at least seems to be functioning. However my logs had copies of this message ranging from line 14538 to line 15452.
```
[Error  : Unity Log] IndexOutOfRangeException: Index was outside the bounds of the array.
Stack trace:
Biodiversity.Creatures.Ogopogo.OgopogoAI+<>c.<Start>b__39_0 (System.UInt64 playerId) (at ./Creatures/Ogopogo/OgopogoAI.cs:179)
Biodiversity.Util.Types.CachedDictionary`2[TKey,TValue].get_Item (TKey key) (at ./Util/Types/CachedDictionary.cs:55)
Biodiversity.Creatures.Ogopogo.OgopogoAI.FixedUpdate () (at ./Creatures/Ogopogo/OgopogoAI.cs:250)
```

I also found this aloe related message being spammed over and over in the log from earlier. This was in Scoopy's CastleFlow on Tolian's Azure.
```
[Error  : Unity Log] IndexOutOfRangeException: Index was outside the bounds of the array.
Stack trace:
Biodiversity.Creatures.Aloe.AloeClient.HandleTargetPlayerChanged (System.UInt64 oldValue, System.UInt64 newValue) (at ./Creatures/Aloe/AloeClient.cs:938)
Unity.Netcode.NetworkVariable`1[T].Set (T value) (at <895801699cfc4b4ab52267f31e2a4998>:IL_001F)
Unity.Netcode.NetworkVariable`1[T].set_Value (T value) (at <895801699cfc4b4ab52267f31e2a4998>:IL_0045)
Biodiversity.Creatures.Aloe.AloeUtils.ChangeNetworkVar[T] (Unity.Netcode.NetworkVariable`1[T] networkVariable, T newValue) (at ./Creatures/Aloe/AloeUtils.cs:508)
Biodiversity.Creatures.Aloe.BehaviourStates.RoamingState+TransitionToPassivelyStalkingPlayer.OnTransition () (at ./Creatures/Aloe/BehaviourStates/RoamingState.cs:123)
Biodiversity.Creatures.Aloe.AloeServer.DoAIInterval () (at ./Creatures/Aloe/AloeServer.cs:205)
(wrapper dynamic-method) EnemyAI.DMD<EnemyAI::Update>(EnemyAI)
Biodiversity.Creatures.Aloe.AloeServer.Update () (at ./Creatures/Aloe/AloeServer.cs:167)
```

**Comment by Ccode-lang** [comment]
Are these using this version of the mod? If so then that might be the problem. The old code did not account for the company cruiser and broke stuff frequently because of the cruiser.

By the way sorry about the long wait time on this issue this repo is no longer in use and we didn't think people were looking at it. The actual project is now open source and stored in another repo in this organization.

**Comment by NotAVerb** [comment]
In my case I was using the latest available from Thunderstore, so I'm pretty sure it was v0.1.4. Should have specified. My bad.

**Comment by Ccode-lang** [comment]
The aloe message will be fixed in the next update. The Ogo error looks like a player joined after Ogopogo spawned. Do you have a late join mod installed?

P.S. It's fine, ogo's code hasn't changed for a while other than for a few bug fixes.

**Comment by NotAVerb** [comment]
I was testing in solo. Likely something to do with the Internship mod. I don't recall what the full list I was running with was. Probably had LobbyControl installed.

**Comment by Ccode-lang** [comment]
Yeah, internship mod should do that also since they count as players. I could probably get something set up to remake that list every few seconds to prevent that error from happening.

**Comment by Ccode-lang** [comment]
Is this fixed now? The newest update should have fixed the error

**Comment by NotAVerb** [comment]
Honestly I won't be able to say anything for certain unless it happens again. I'd tentatively mark as solved. Only thing to do is wait and see if someone ever reports something similar.

**Comment by Ccode-lang** [comment]
Thank you for telling us about the issue. I'll reopen it if it still turns out to be a problem.
